types/Accounts.ts

```diff
@@ -56,76 +32,34 @@ export type TokenInfo = {
   refreshToken?: string;
 };
 
-export interface PersonalAccessKeyAccount_NEW extends CLIAccount_NEW {
-  authType: 'personalaccesskey';
-  personalAccessKey: string;
-}
-
-export interface PersonalAccessKeyAccount_DEPRECATED
-  extends CLIAccount_DEPRECATED {
-  authType: 'personalaccesskey';
+export interface PersonalAccessKeyConfigAccount
+  extends BaseHubSpotConfigAccount {
+  authType: typeof PERSONAL_ACCESS_KEY_AUTH_METHOD.value;
   personalAccessKey: string;
-}
-
-export type PersonalAccessKeyAccount =
-  | PersonalAccessKeyAccount_NEW
-  | PersonalAccessKeyAccount_DEPRECATED;
-
-export interface OAuthAccount_NEW extends CLIAccount_NEW {
-  authType: 'oauth2';
   auth: {
-    clientId?: string;
-    clientSecret?: string;
-    scopes?: Array<string>;
-    tokenInfo?: TokenInfo;
+    tokenInfo: TokenInfo;
   };
 }
 
-export interface OAuthAccount_DEPRECATED extends CLIAccount_DEPRECATED {
-  authType: 'oauth2';
+export interface OAuthConfigAccount extends BaseHubSpotConfigAccount {
+  authType: typeof OAUTH_AUTH_METHOD.value;
   auth: {
-    clientId?: string;
-    clientSecret?: string;
-    scopes?: Array<string>;
-    tokenInfo?: TokenInfo;
+    clientId: string;
+    clientSecret: string;
+    scopes: Array<string>;
+    tokenInfo: TokenInfo;
   };
 }
 
-export type OAuthAccount = OAuthAccount_NEW | OAuthAccount_DEPRECATED;
-
-export interface APIKeyAccount_NEW extends CLIAccount_NEW {
-  authType: 'apikey';
+export interface APIKeyConfigAccount extends BaseHubSpotConfigAccount {
+  authType: typeof API_KEY_AUTH_METHOD.value;
   apiKey: string;
 }
 
-export interface APIKeyAccount_DEPRECATED extends CLIAccount_DEPRECATED {
-  authType: 'apikey';
-  apiKey: string;
-}
-
-export type APIKeyAccount = APIKeyAccount_NEW | APIKeyAccount_DEPRECATED;
-
-export interface FlatAccountFields_NEW extends CLIAccount_NEW {
-  tokenInfo?: TokenInfo;
-  clientId?: string;
-  clientSecret?: string;
-  scopes?: Array<string>;
-  apiKey?: string;
-  personalAccessKey?: string;
-}
-
-export interface FlatAccountFields_DEPRECATED extends CLIAccount_DEPRECATED {
-  tokenInfo?: TokenInfo;
-  clientId?: string;
-  clientSecret?: string;
-  scopes?: Array<string>;
-  apiKey?: string;
-  personalAccessKey?: string;
-}
-
-export type FlatAccountFields =
-  | FlatAccountFields_NEW
-  | FlatAccountFields_DEPRECATED;
+export type HubSpotConfigAccount =
```

Comment:
This a) lets us confirm that accounts returned by config methods have valid auth and b) lets us use the `accountType` field to determine exactly what type an account is